File: erykandroidcommon/src/androidTest/java/pl/edu/amu/wmi/erykandroidcommon/di/TestModule.java
package pl.edu.amu.wmi.erykandroidcommon.di;

import android.app.Application;

import org.mockito.Mockito;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import pl.edu.amu.wmi.erykandroidcommon.service.UserService;

@Module
public class TestModule {

    private final Application mApplication;

    public TestModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    protected UserService provideUserService() {
        return Mockito.mock(UserService.class);
    }

}

File: erykandroidcommon/src/androidTest/java/pl/edu/amu/wmi/erykandroidcommon/di/TestModule.kt
package pl.edu.amu.wmi.erykandroidcommon.di

import android.app.Application

import org.mockito.Mockito

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import pl.edu.amu.wmi.erykandroidcommon.service.UserService

@Module
class TestModule(private val mApplication: Application) {

    @Provides
    @Singleton
    fun provideUserService(): UserService<*, *> {
        return Mockito.mock(UserService<*, *>::class.java)
    }

}


